### PR TITLE
Initial commit of smartd role

### PR DESCRIPTION
**Pull request by SlothOfAnarchy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# smartd
+
+This role installs and configures `smartd`. 
+In the default configuration, all devices will be tested and observed, but no mails will be sent.
+
+## Requirements
+
+A Debian or Arch based distribution.
```
Comment (ghost) [inline README.md]: You don't mentian Debian in the Ansible metadata

File: README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+`/usr/bin/mail` has to exist and be configured to be able to send report mails.
```
Comment (ghost) [inline README.md]: *if report mails are desired

File: README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## Role Variables
+
+This role only has one top-level var called `smartd_devices`, which is a dict of dicts to specify device monitoring.
```
Comment (ghost) [inline README.md]: The role doesn't "have" the var, it expects/requires/... it

File: README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Each key corresponds to one drive path, while `DEVICESCAN` (the only default entry) may be used to scan for devices.
+After it, all further lines are ignored by smartd, so you probably want to insert it at the end or not at all.
```
Comment (ghost) [inline README.md]: Huh?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+|:-----------------------------|:-------------------------------------------------:|:--------------------------------------------------------------------------------------------------------------------------------------------|
+| `type`                       |                      `auto`                       | The type of device, e.g. `ata`, `scsi`, `marvell`, etc.                                                                                     |
+| `nocheck`                    |                  `standby,15,q`                   | Powermode setting. By default, devices in `standby` are only woken up every 15th attempt and skipped check attempts are not logged.         |
+| `check_type`                 |                     `normal`                      | Use `normal` or `permissive` to force SMART checking even if it is not advertised.                                                          |
+| `ata_offline_testing`        |                                                   | Set to `on` or `off` to configure ATA offline testing on smartd startup.                                                                    |
```
Comment (ghost) [inline README.md]: Isn't there a default?

File: defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+smartd_devices:
```
Comment (ghost) [inline defaults/main.yml:2]: Where are my three dashes?

File: meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+galaxy_info:
```
Comment (ghost) [inline meta/main.yml:2]: Where are my three dashes?

File: meta/main.yml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  author: Insert author(s) here
```
Comment (ghost) [inline meta/main.yml]: That's not correct, also correct the rest of the file

File: meta/main.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  platforms:
+    - name: Ubuntu
+      versions:
+        - xenial
+        - yakkety
+        - zesty
+        - artful
+
+  galaxy_tags:
+    - stuvus
```
Comment (ghost) [inline meta/main.yml]: Old metadata?

File: tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+  when: "ansible_pkg_mgr == 'apt'"
+
+- name: Update repository cache (pacman)
+  pacman:
+    update_cache: yes
+  changed_when: false
+  when: "ansible_pkg_mgr == 'pacman'"
+
+- name: Install smartmontools (apt)
+  apt:
```
Comment (ghost) [inline tasks/main.yml]: Maybe use "package" instead of "apt" and pacman"? Might cover more distributions

File: tasks/main.yml [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  
+- name: Create smartd configuration
+  template:
+    src: smartd.conf.j2
+    dest: /etc/smartd.conf
+    owner: root
+    group: root
+    mode: 0644
+    backup: yes
+  notify: Restart smartd
```
Comment (ghost) [inline tasks/main.yml]: EOL missing

**Review by ghost — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (SlothOfAnarchy) [reply README.md]: Yes, it *uses* or *needs* the var, though it is already set through the defaults.

File: README.md (hunk 4 above)
Comment (SlothOfAnarchy) [reply README.md]: See the linked man page:
> If a non-comment entry in the configuration file is the text string DEVICESCAN in capital letters, then smartd will ignore any remaining lines in the configuration file, and will scan for devices.

File: README.md (hunk 5 above)
Comment (SlothOfAnarchy) [reply README.md]: No there isn't. If this option is set, an ATA command will be issued at smartd startup which configures the device to enable or disable offline testing. This setting stays the same through reboots. If this var is not set, the mode on the device won't be changed.

File: defaults/main.yml (hunk 6 above)
Comment (SlothOfAnarchy) [reply defaults/main.yml:2]: I would've bet you'd think of these as optional bloat. I'll be sure to add them ;)

File: meta/main.yml (hunk 8 above)
Comment (SlothOfAnarchy) [reply meta/main.yml]: Must've forgot to fill it out.

File: meta/main.yml (hunk 9 above)
Comment (SlothOfAnarchy) [reply meta/main.yml]: Indeed

File: tasks/main.yml (hunk 10 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: I think it's obvious that `smartmontools` is a package and it is being installed using apt (in this case).

File: tasks/main.yml (hunk 11 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml]: hair-splitting

**Comment by SlothOfAnarchy** [comment]
@dasJ @schuppentier please rereview.

File: README.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+| `type`                       |                      `auto`                       | The type of device, e.g. `ata`, `scsi`, `marvell`, etc.                                                                                                                           |
+| `nocheck`                    |                  `standby,15,q`                   | Powermode setting. By default, devices in `standby` are only woken up every 15th attempt and skipped check attempts are not logged.                                               |
+| `check_type`                 |                     `normal`                      | Use `normal` or `permissive` to force SMART checking even if it is not advertised.                                                                                                |
+| `ata_offline_testing`        |                                                   | Set to `on` or `off` to enable or disable automatic ATA offline testing by the device itself. Note that you have to quote the value because of ansible's interpretation of `off`. |
```
Comment (dasJ) [inline README.md]: So does this have a default now?

File: meta/main.yml [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+        - trusty
+        - xenial
+        - yakkety
+        - zesty
+        - artful
+    - name: ArchLinux
+      versions:
+        - all
+        - any
+    - name: Debian
```
Comment (dasJ) [inline meta/main.yml]: Is this even tested against Debian?

**Review by dasJ — CHANGES_REQUESTED** [review]

File: tasks/main.yml (hunk 10 above)
Comment (dasJ) [reply tasks/main.yml]: You didn't get my point, but it doesn't matter

File: meta/main.yml (hunk 13 above)
Comment (SlothOfAnarchy) [reply meta/main.yml]: I tested it on all my debian machines.

Comment (dasJ) [reply meta/main.yml]: You don't own any debian machines

Comment (SlothOfAnarchy) [reply meta/main.yml]: Did you just assume my operating system?

Comment (dasJ) [reply meta/main.yml]: Yes. Just fix this issue, it's not that complicated

**Review by dasJ — APPROVED** [review]

**Pull request merged**